docs/cart-and-checkout-blocks/hooks/hook-alternatives.md

```diff
@@ -0,0 +1,456 @@
+    - This is supported by our use of Slot/Fill and by registering inner blocks as children of the block you want to show them after.
+- Adding payment methods
+    - Simply adding the payment method is not enough, developers need to register a React component to show in the payments block as well.
+- Adding shipping methods for local pickup
+    - Should work as normal, but does require the method to support collectable
+- Modify the names of items in the cart
+    - Requires the developer to use a JS filter rather than PHP filter
+
+#### Not supported at all
+
+- Modifying individual cart line items to add or change content/change HTML (outside of what is offered by our checkout filters).
+
+### Checkout block
+
+#### Fully supported with no modification needed
+
+- Pre-filling values on the form (Some merchants achieve this by overriding their values using `woocommerce_checkout_get_value`)
+    - This is supported insofar as we respect what has been set on the cart/customer object when loading the cart. `woocommerce_checkout_get_value` is not supported in the Checkout block, but this is this hooks only use-case.
+
+#### Partially supported or supported by alternative approach
+
+- Adding new Checkout fields (commonly achieved with the `woocommerce_checkout_fields` hook)
+    - Supported by the Additional Checkout Fields API, not all field types are supported right now
+- Custom field validation using hooks such as woocommerce_checkout_process
+    - Supported by hooking into the payment processing events - validation at the point of pressing "Place order" this is only be possible for payment methods. Other extensions could add validation errors to the data store prior to the button being pressed, and this would prevent checkout, but it is not a good solution, especially if the validation checks are expensive.
+- Updating orders/customers with hooks before/during/after order placement
+    - We have some hooks that fire during checkout:
+        - `woocommerce_store_api_checkout_order_processed` - fired when the order has been placed 
+        - `woocommerce_store_api_checkout_update_customer_from_request` - fired when the customer is being updated with data from the checkout request 
+        - `woocommerce_store_api_checkout_update_order_meta` - fired when order meta data is being updated with data from the request
+    - Any other WC hooks that fire in the Shortcode process (e.g. `woocommerce_checkout_order_processed` will not fire on Store API requests from the blocks)
+- Customising the place order button text using `woocommerce_order_button_text` 
+    - Not supported with hooks but it is possible in the editor. There is no programmatic way to achieve this right now.
+
+#### Not supported at all
+
+- Changing existing core fields (commonly achieved with the `woocommerce_checkout_fields` hook)
```

Comment:
Thought visibility/labels/position can be changed on a country per country basis

Reply:
You have to use locale for this, checkout_fields doesn't work.